src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -752,7 +750,30 @@ protected override int ExecuteCmd(DocumentsOperationContext context)
                             DocumentsStorage.PutOperationResults putResult;
                             try
                             {
-                                putResult = Database.DocumentsStorage.Put(context, cmd.Id, cmd.ChangeVector, cmd.Document);
+                                var flags = DocumentFlags.None;
+                                
+                                //if (cmd.ForceRevision)
+                                if (cmd.ForceRevisionCreationStrategy == ForceRevisionStrategy.Before)
+                                // Note: we currently only handle 'Before'. 
+                                // Creating the revision 'After' will be done only upon customer demand.
+                                {
+                                    var existingDocument = Database.DocumentsStorage.Get(context, cmd.Id);
+                                    if (existingDocument == null)
+                                    {
```

Comment:
Am I missing something, the comment on `HandleForceRevisionCreation` says that we can force revision on document creation, but here it will throw?

Reply:
1. Comment here means: 
We cannot force/create a Revision if the document (for which a Forced Revision was requested) was Not yet saved to the server.

2. Comment in `HandleForceRevisionCreation` means:
The original document (for which you request a Forced Revision) may have changed & PUT to the server with this flow:

* Create document
* Save document to server (document is created without the HasRevisions Flag)
* Request a Forced Revision
   => A forced revision is created
   => Original document is modified and saved - since HasRevisions Flag is added to it 
         So for tracked entities - `HandleForceRevisionCreation` must update the changeVector in the session 

Maybe I should rephrase so its clearer